kingpin/test/test_utils.py

```diff
@@ -33,6 +33,20 @@ def test_populate_with_env(self):
         result = utils.populate_with_tokens(string, tokens)
         self.assertEquals(result, expect)
 
+    def test_populate_with_values_not_default(self):
+        tokens = {'UNIT_TEST': 'FOOBAR', 'SECOND_UNIT': 'BARBAR'}
+        string = 'Unit %UNIT_TEST?DEFAULT% Test %SECOND_UNIT?DEFAULT2%'
```

Comment:
I suggest just a single `|` ... using `?=` is too bash-like. A semi-colon would be OK too I suppose, but I feel like `|` really means `or` to most people.